### ISSUE TITLE
NaNs are ignored for mean calculation but not variance

### ISSUE THREAD

**Issue by joezuntz** [rmjarvis/TreeCorr#90]
If you pass in g1 or g2 arrays containing NaNs, TreeCorr warns you and tells you that it's ignoring the affected rows.

This is true for the returned mean values xip and xim, but not for the variances varxip and varxim, which come back as NaNs if there are any NaN's in the input.

Here's an example:
```
import numpy as np
import treecorr

ra = np.random.uniform(0, 20, 10000)
dec = np.random.uniform(0, 20, 10000)

g1 = np.random.uniform(-0.05, 0.05, 10000)
g2 = np.random.uniform(-0.05, 0.05, 10000)

def calculate(ra, dec, g1, g2):
    config = {
        'min_sep': 0.5,
        'max_sep': 60.0,
        'nbins': 10,
        'bin_slop':0.01,
        'sep_units':'arcmin',        
    }

    gg = treecorr.GGCorrelation(config)
    cat = treecorr.Catalog(ra=ra, dec=dec, g1=g1, g2=g2, ra_units='deg', dec_units='deg')
    gg.process(cat)
    return gg.xip, gg.varxip


mu1, var1 = calculate(ra, dec, g1, g2)

# Now add some Nan's to the end of the data
# TreeCorr's message warns about this but says it's ignoring the rows
ra = np.concatenate((ra, [0.0]))
dec = np.concatenate((dec, [0.0]))
g1 = np.concatenate((g1, [np.nan]))
g2 = np.concatenate((g2, [np.nan]))


mu2, var2 = calculate(ra, dec, g1, g2)

# Passes - NaNs ignored in mean calculation
assert np.all(mu1==mu2)

# Fails, var2 is NaNs
assert np.all(var1==var2)
```

**Comment by rmjarvis** [comment]
Thanks Joe.  And thanks for providing the appropriate unit test for me.  :)

The bug was pretty simple.  In the varg (and vark) calculation, I was just summing w * g**2.  I changed that to only sum elements with w != 0, which avoids the ones with NaN g1,g2 values, so the NaN doesn't propagate to the sum.  

I looked over the rest of the input code and I don't see any other places where the code assumed that multiplying by w was sufficient to skip the entries I wanted to skip.  As you noted, the code for doing the correlations was already checking for w=0 and skipping those entries.

cf. PR #91
